src/libmongoc/src/mongoc/mongoc-client-side-encryption.c

```diff
@@ -1038,6 +1226,50 @@ append_bson_range_opts(bson_t *bson_range_opts, const mongoc_client_encryption_e
    }
 }
 
+static void
+append_bson_text_per_index_opts(bson_t *out, const struct _encrypt_text_per_index_opts_t *opts)
+{
+   BSON_ASSERT_PARAM(out);
+   if (opts->str_max_length.set) {
+      BSON_ASSERT(bson_append_int32(out, "strMaxLength", -1, opts->str_max_length.value));
+   }
+   if (opts->str_max_query_length.set) {
+      BSON_ASSERT(bson_append_int32(out, "strMaxQueryLength", -1, opts->str_max_query_length.value));
+   }
+   if (opts->str_min_query_length.set) {
+      BSON_ASSERT(bson_append_int32(out, "strMinQueryLength", -1, opts->str_min_query_length.value));
+   }
+}
+
+static void
+append_bson_text_opts(bson_t *bson_text_opts, const mongoc_encrypt_text_opts_t *opts)
+{
+   BSON_ASSERT_PARAM(bson_text_opts);
+   BSON_ASSERT_PARAM(opts);
+
+   BSON_ASSERT(BSON_APPEND_BOOL(bson_text_opts, "caseSensitive", opts->case_sensitive));
```

Comment:
I expect `caseSensitive` and `diacriticSensitive` should only be applied if set, like:

```c
if (opts->diacritic_sensitive.set) {
    BSON_ASSERT(BSON_APPEND_BOOL(bson_text_opts, "diacriticSensitive", opts->diacritic_sensitive.value));
}
```

The current implementation defaults to `false`. The spec requires these be set. If a user does not set either field, I expect libmongocrypt [will error](https://github.com/mongodb/libmongocrypt/blob/f8dba59031ca9dbce87c4829f9c9c4eb9678e870/src/mc-textopts.c#L208-L211).